code/Controllers/CMSMain.php

```diff
@@ -1038,56 +1038,93 @@ public function getBreadcrumbsBackLink()
+                    _t(CMSMain::class . '.SEARCHRESULTS', 'Search results'),
+                    ($unlinked) ? false : $this->LinkPages()
+                );
+            } elseif ($parentID = $this->getRequest()->getVar('ParentID')) {
+                // We're navigating the listview. ParentID is the page whose
+                // children are currently displayed.
+                if ($page = SiteTree::get()->byID($parentID)) {
+                    $this->buildListViewBreadcrumb($items, $page);
+                }
+            }
         }
 
-        // Check if we are editing a page
-        /** @var SiteTree $record */
-        $record = $this->currentPage();
-        if (!$record) {
-            $items->push(new ArrayData([
-                'Title' => CMSPagesController::menu_title(),
-                'Link' => ($unlinked) ? false : $this->LinkPages()
-            ]));
+        $this->extend('updateBreadcrumbs', $items);
 
-            $this->extend('updateBreadcrumbs', $items);
+        return $items;
+    }
 
-            return $items;
-        }
+    /**
+     * Push the provided an extra breadcrumb crumb at the end of the provided List
+     */
+    private function pushCrumb(ArrayList $items, string $title, string|false $link): void
+    {
+        $items->push(ArrayData::create([
+            'Title' => $title,
+            'Link' => $link
+        ]));
+    }
 
-        // Add all ancestors
-        $ancestors = $record->getAncestors();
-        $ancestors = new ArrayList(array_reverse($ancestors->toArray() ?? []));
-        $ancestors->push($record);
-        /** @var SiteTree $ancestor */
+    /**
+     * Build Breadcrumb for the Edit page form. Each crumb links back to its own edit form.
+     */
+    private function buildEditFormBreadcrumb(ArrayList $items, SiteTree $page, bool $unlinked): void
+    {
+        // Find all ancestors of the provided page
+        $ancestors = $page->getAncestors(true);
+        $ancestors = array_reverse($ancestors->toArray() ?? []);
         foreach ($ancestors as $ancestor) {
-            $items->push(new ArrayData([
-                'Title' => $ancestor->getMenuTitle(),
-                'Link' => ($unlinked)
-                    ? false
-                    : $ancestor->CMSEditLink()
-            ]));
+            // Link to the ancestor's edit form
+            $this->pushCrumb(
+                $items,
+                $ancestor->getMenuTitle(),
+                $unlinked ? false : $ancestor->CMSEditLink()
+            );
         }
+    }
 
-        $this->extend('updateBreadcrumbs', $items);
-
-        return $items;
+    /**
+     * Build Breadcrumb for the List view. Each crumb links to the list view for that page.
+     */
+    private function buildListViewBreadcrumb(ArrayList $items, SiteTree $page): void
+    {
+        // Find all ancestors of the provided page
+        $ancestors = $page->getAncestors(true);
+        $ancestors = array_reverse($ancestors->toArray() ?? []);
```

Comment:
This is very slightly less code than the original, but you're looping over everything twice now instead of just once! No array_unshift though so it maaaaybe balances out? Probably negligible anyway so I'm okay with it.